main.go

```diff
@@ -325,6 +331,57 @@ func (s *service) openNewIssues(issueMap map[string]*github.Issue, comments []*t
 	log.Printf("Created new issues. count=%v", count)
 }
 
+func (s *service) assignNewIssues(issueMap map[string]*github.Issue, assigneeMap *map[string]string) {
+	for title, assignee := range *assigneeMap {
+		issue := issueMap[title]
+		issueNumber := issue.GetNumber()
+		req := &github.IssueRequest{
+			Assignees: &[]string{assignee},
+		}
+		_, _, err := s.client.Issues.Edit(s.ctx, s.env.owner, s.env.repo, issueNumber, req)
+		if err != nil {
+			log.Printf("Error while assigning %v to issue %v. err=%v", assignee, issueNumber, err)
+		} else {
+			log.Printf("Successfully assigned %v to issue %v.", assignee, issueNumber)
+		}
+	}
+}
+
+func (s *service) getAssigneeFromCommitHash(commitHash string, title string, assigneeMap *map[string]string, successCount *int, assigneeMapMux sync.Locker, wg *sync.WaitGroup) {
+	defer wg.Done()
+	commit, _, err := s.client.Repositories.GetCommit(s.ctx, s.env.owner, s.env.repo, commitHash, &github.ListOptions{})
+	if err != nil {
+		log.Printf("Error while getting commit from commit hash. err=%v", err)
+	} else {
+		assigneeMapMux.Lock()
+		defer assigneeMapMux.Unlock()
```

Comment:
For short code blocks you don't need `defer` to unlock the mutex. It's more fail-safe, since if you did that in a loop, it would be a deadlock with `defer`